### ISSUE TITLE
Document the error: system DNS configuration could not be read

### ISSUE THREAD

**Issue by dcorking** [cloudflare/cloudflare-docs#13004]
### Which Cloudflare product(s) does this pertain to?

WARP Client
Version: 2024.1.160.0 (20240124.7)

### Subject Matter

Unable to connect reason system DNS configuration could not be read

### Content Location

new section in https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/troubleshooting/client-errors/

### Additional information

On a Mac I get this error:
```
$ warp-cli status
Status update: Unable to connect. Reason: The system DNS configuration could not be read. If on macOS / Linux, try restoring /etc/resolv.conf to default
Success
```
I'd like to request documentation that explains more detail about how to restore this file to the default, and what to do if the error persists when the file has the default contents.

As seen on macOS Ventura 13.6.4.
However everything seems normal and not knowingly changed from when WARP Client used to connect successfully:
```bash
$ cat /etc/resolv.conf
#
# macOS Notice
#
# This file is not consulted for DNS hostname resolution, address
# resolution, or the DNS query routing mechanism used by most
# processes on this system.
#
# To view the DNS configuration used by this system, use:
#   scutil --dns
#
# SEE ALSO
#   dns-sd(1), scutil(8)
#
# This file is automatically generated.
#
nameserver 1.0.0.01
nameserver 1.1.1.01
$ scutil --dns
DNS configuration

resolver #1
  nameserver[0] : 1.0.0.1
  nameserver[1] : 1.1.1.1
  flags    : Request A records
  reach    : 0x00000002 (Reachable)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #3
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #4
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #5
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #6
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #7
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

resolver #8
  domain   : test
  nameserver[0] : 127.0.0.1
  port     : 9253
  flags    : Request A records, Request AAAA records
  reach    : 0x00030002 (Reachable,Local Address,Directly Reachable Address)

DNS configuration (for scoped queries)

resolver #1
  nameserver[0] : 1.0.0.1
  nameserver[1] : 1.1.1.1
  if_index : 19 (en7)
  flags    : Scoped, Request A records
  reach    : 0x00000002 (Reachable)
$
```

**Comment by ranbel** [comment]
@dh-cf for guidance on 

> how to restore this file to the default, and what to do if the error persists when the file has the default contents.

**Comment by dcorking** [comment]
From experience, one workaround you _could_ consider documenting is simply to try again later. 

### more info

I tried again later in the day I wrote my report, and the client successfully started an IPv4 connection to WARP, and I am using that connection to post this comment. I am sorry I can't provide you any diagnostics regarding what changed, as I don't know. As far as I can tell, when the client connects, macOS successfully rewrites /etc/resolv.conf to point to 127.0.2.2 and 127.0.2.3, which is also reflected in the output of scutil --dns
```shell
$ cat /etc/resolv.conf
#
# macOS Notice
#
# This file is not consulted for DNS hostname resolution, address
# resolution, or the DNS query routing mechanism used by most
# processes on this system.
#
# To view the DNS configuration used by this system, use:
#   scutil --dns
#
# SEE ALSO
#   dns-sd(1), scutil(8)
#
# This file is automatically generated.
#
nameserver 127.0.2.2
nameserver 127.0.2.3
$ scutil --dns
DNS configuration

resolver #1
  nameserver[0] : 127.0.2.2
  nameserver[1] : 127.0.2.3
  flags    : Request A records
  reach    : 0x00030002 (Reachable,Local Address,Directly Reachable Address)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #3
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #4
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #5
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #6
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #7
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

resolver #8
  domain   : test
  nameserver[0] : 127.0.0.1
  port     : 9253
  flags    : Request A records, Request AAAA records
  reach    : 0x00030002 (Reachable,Local Address,Directly Reachable Address)

DNS configuration (for scoped queries)

resolver #1
  nameserver[0] : 127.0.2.2
  nameserver[1] : 127.0.2.3
  if_index : 19 (en7)
  flags    : Scoped, Request A records
  reach    : 0x00030002 (Reachable,Local Address,Directly Reachable Address)

resolver #2
  nameserver[0] : 127.0.2.2
  nameserver[1] : 127.0.2.3
  if_index : 20 (utun3)
  flags    : Scoped, Request A records, Request AAAA records
  reach    : 0x00030002 (Reachable,Local Address,Directly Reachable Address)
$
```

**Comment by ranbel** [comment]
Resolved in #14702
